auth0_flutter/FAQ.md

```diff
@@ -145,14 +144,64 @@ Otherwise, the browser modal will close right away and the user will be automati
+
+- When not using refresh tokens, the SPA SDK relies on third-party cookie support to log in silently. If you're not using refresh tokens, you could be using a browser that blocks third-party cookies by default (Safari, Brave, etc)
+- You're using the [classic login experience](https://auth0.com/docs/authenticate/login/auth0-universal-login/classic-experience), and trying to log in using a social provider that uses Auth0's developer keys (see [Limitations of Developer Keys when using Classic Universal Login](https://auth0.com/docs/authenticate/identity-providers/social-identity-providers/devkeys#limitations-of-developer-keys-when-using-universal-login))
+
+Please try these to see if you can get unblocked:
+
+- Try it in a browser like Chrome which does not block third-party cookies by default (yet)
+- Use the New Login Experience, if possible
+- Supply the social connection with your own client ID and secret in the Auth0 dashboard
+
+#### Using Multi-factor Authentication (MFA)
+
+By default, `auth0-spa-js` uses the `prompt=none` and `response_mode=web_message` flow for silent auth, which depends on the user's Auth0 session.
+
+If you have "Require Multi-factor Auth" set to "Always" in your [Auth0 Dashboard](https://manage.auth0.com/#/security/mfa), silent authentication from your SPA will fail unless:
+
+- The user is using a one-time code and selects "Remember me for 30 days"
+- `allowRememberBrowser` is [configured in a Rule](https://auth0.com/docs/secure/multi-factor-authentication/customize-mfa#change-authentication-request-frequency) and `provider` is set to `google-authenticator`
+
+If silent auth is being used and Auth0 needs interaction from the user to complete the MFA step, then authentication will fail with an `mfa_required` error and the user must log in interactively.
+
+To resolve this:
+
+- Use a Rule to [configure the MFA step](https://auth0.com/docs/secure/multi-factor-authentication/customize-mfa#change-authentication-request-frequency) to allow the user to remember the browser for up to 30 days
+- Use refresh tokens + local storage so that the auto-login on page refresh does not depend on the user's session. Please [read the docs on storage options](https://auth0.com/docs/libraries/auth0-single-page-app-sdk#change-storage-options) and [rotating refresh tokens](https://auth0.com/docs/libraries/auth0-single-page-app-sdk#use-rotating-refresh-tokens) as these change the security profile of your application.
+
+### 2. Why do I get `auth0-spa-js must run on a secure origin`?
+
+Internally, the SPA SDK uses [Web Cryptography API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API) to create [SHA-256 digest](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest).
+
+According to the spec ([via Github issues](https://github.com/w3c/webcrypto/issues/28)), Web Cryptography API requires a secure origin, so that accessing `Crypto.subtle` in a not secure context return undefined.
```

Comment:
```suggestion
According to the spec ([via Github issues](https://github.com/w3c/webcrypto/issues/28)), Web Cryptography API requires a secure origin, so that accessing `Crypto.subtle` in a not secure context returns `undefined`.
```